dpctl/tests/test_usm_ndarray_indexing.py

```diff
@@ -252,8 +253,14 @@ def test_advanced_slice5():
     q = get_queue_or_skip()
     ii = dpt.asarray([1, 2], sycl_queue=q)
     x = _make_3d("i4", q)
-    with pytest.raises(IndexError):
-        x[ii, 0, ii]
+    y = x[ii, 0, ii]
```

Comment:
Do we need to have any test, where we are testing different combination of dtypes for indices arrays?

Reply:
Such tests were already added, just only using integer arrays (see: `test_advanced_integer_indexing_cast_indices`). I don't think we need one for Python scalars because `dpt.asarray` is called on them, followed by the same coercion as in the array case.

Reply:
I think it would be excessive to add boolean to it, since `nonzero` converts to integer anyway. I think adding a comment/documenting it in `_prepare_indices_arrays` makes a lot of sense though.